### DISCUSSION TITLE
Implement application locking with either pin or biometric support

### DISCUSSION THREAD

**Discussion by martijndeb** [pachli/pachli-android#264] [Ideas]
To add an extra layer of security, it would be great if the application can support either PIN locking or biometric locking. (Like for example your banking app, or apps like Proton Mail).

This has the following benefits added after device locking;
- When using the app on a shared device, like for instance a family tablet, you can close off the app. For now you will have to resort to Androids multi user mode, which has it's own set of problems.
- When your device is lost, and/or unlocked, you have a secondary layer of protection (given you don't use the same method as picked for device locking). This helps safeguard misusing your account.

Safe guarding account usage is especially important when having moderation/administration access to instances.

**Comment by nikclayton** [comment]
Thanks for the suggestion, I can see how that would be useful.

How often would Pachli need to reprompt for the pin/biometric unlock?

For example, only when resuming Pachli after the screen has been turned off?

Or whenever you return to Pachli (e.g., opening a link from Pachli in a browser, and then pressing the back button to return to Pachli)?

Just to do some expectation setting -- I doubt this'll get worked on until Pachli has better UX testing that can ensure that this sort of functionality is working as intended. This feels like something that it would be easy to inadvertently break under some scenarios, and not notice that during manual testing.

**Reply by martijndeb** [reply]
I think that most apps use a timeout for this. That's set to something like 5 minutes. Something like that is okay with me.

I'm also okay with whatever timeline this needs, it's a feature i'd really like but it's also just an improvement.
